### ISSUE TITLE
[UX][DX] When a module is enabled, we should set a message containing the link to configure

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#5135]
**Description of the need**

I've noticed that in a handful of contrib modules, there is an install hook that simply sets a message with a link to the configuration page. In Drupal, I love this feature as it saves me from needing to find the module that was just installed, open it's collapsed set of buttons, and click it's configuration link.

Here's some example code from the sendgrid integration module:

```php
function sendgrid_integration_install() {
  $t = get_t();
  backdrop_set_message($t('You can configure SendGrid settings under Administer > Configuration > !link', ['!link' => l($t('SendGrid settings'), 'admin/config/system/sendgrid'),]));
}
```

**Proposed solution**

Since modules already define a configuration link in the .info file, we could use that link to set this message automatically for all projects.

[DX] This would improve the experience for contributed module developers, as they wouldn't need to repeat the same action for every module they build.

[UX] I believe this would also be a user experience improvement, but I'd love more feedback from others.

**Alternatives that have been considered**

In Backdrop, we've already improved the module installation scenario by 1) remembering the filter for the module that was just installed, if you'd used it before installing the module it remains after -- making the actions links immediately available. And 2) we've removed the collapsed set of buttons, instead replacing them with action links that are more immediately available.

**Additional information**

TBD 

**Draft of feature description for Press Release (1 paragraph at most)**

When installing a module in Backdrop, you'll be immediately be presented with a link to the module's configuration page, if it has one.

**Comment by indigoxela** [comment]
Sounds reasonable, we currently only show "Installation is complete." An addition to that message with a link to the config form - if any - saves some clicks.

How about the handful of modules, that already provide a message in their hook_install? The maintainers should then remove these to avoid duplication, but how would they know?

**Comment by jenlampton** [comment]
If this improvement went into a minor version instead of a bug-fix version we'd be able to do a little more PR around the change. We could also search contrib for a hook_install and see what projects we have that do it, like so: https://github.com/search?utf8=%E2%9C%93&q=hook_install+in:file+org:backdrop-contrib&type=Code

**Comment by findlabnet** [comment]
How will it look like when several modules installed / enabled at the same time?

**Comment by jenlampton** [comment]
I was thinking each module would get its own message, much like happens now if you were to enable twilio and hax at the same time.

**Comment by philsward** [comment]
Maybe set a limit on individual say 4 and if more than 4, the message status changes to something like "X modules were installed. View list of installed modules..." Linking to a collapsible fieldset to expand  the full list on click.

**Comment by indigoxela** [comment]
Something else came to my mind: some modules require configuration to be useful, some work out-of-the-box and the config form is only there to do some optional tweaks. Not sure if this is worth considering...

**Comment by docwilmot** [comment]
@indigoxela makes a good point. If a module dev wants you to immediately go to config, she'll add a message in `hook_install()`, otherwise leave configuration link discoverable on the modules page.

**Comment by jenlampton** [comment]
> Not sure if this is worth considering...

I don't think this is worth worrying about. The number of current contrib projects where this distinction was made is small. Many modules that do not need to be configured include the message in the install hook, simply because there is a settings page, and many more where configuration is required don't bother to, because the instructions are in the README. Right now, the link appearing is arbitrary. 

It's also a bit of a gray area whether a module *really* needs to be configured or not. Many do, but then also have default values. Is it important to review those settings before using the module? Depends who you ask :) 

Though the best-case scenario is that all developers do carefully decide whether the link should be shown or not, I don't think that's likely to happen. 

We could hope for a best-case scenario and leave things as they are, as @docwilmot suggests, or we could add this code that would make it easier on all contrib developers, and *probably* also easier on site architects.

**Comment by BWPanda** [comment]
> I've noticed that in a handful of contrib modules, there is an install hook that simply sets a message with a link to the configuration page. In Drupal, I love this feature...

I *hate* this 'feature'. I find it obnoxious that modules are trying to shove things in my face. There are links next to each module for configuring it already, and I can also find it in the admin menu if needbe. I don't want/need messages popping up telling me what to configure.

At most, I'd be willing to support an addition to the "Installation complete" message saying that configuration links can be found next to each module below. Since the links only appear after the modules are enabled, I can see how people might miss them and so this could be helpful.

But I'm not in favour of adding a message/link for *each* enabled module...
